media/js/firefox/new-ios-redirect-helper.js

```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// create namespace
+if (typeof window.Mozilla === 'undefined') {
+    window.Mozilla = {};
+}
+
+(function(Mozilla) {
+    'use strict';
+
+    Mozilla.FirefoxNewIosRedirectHelper = {
+        getDomainFromURL: function(url) {
+            var matches = url.match(/^https?\:\/\/([^\/?#]+)(?:[\/?#]|$)/i);
```

Comment:
Regex isn't one of my strong points, but out of curiosity what is the non-capturing group `(?:[\/?#]|$)` needed for? It doesn't seem to be an issue, but the tests seem to pass both with and without it.